File: project-euler/ruby/7.rb
#!/usr/bin/env ruby

def prime(n)
	(2..n**0.5).each do |a|
		return false if n % a == 0
	end
	return true
end

c = 0
n = 2

until c > 10000
	c += 1 if prime(n)
	n += 1
end

puts n-1
# 104743

File: project-euler/python/7.py
#!/usr/bin/env python

from lib import divisible_by_any

# 10,001st prime number

def nth_prime(N):
	primes = [2]
	N -= 1
	x = 3
	while N > 0:
		if not divisible_by_any(x, primes):
			primes.append(x)
			N -= 1
		x += 2
	return primes[-1]


if __name__ == '__main__':
	N = 10001
	print "Answer: " + str(nth_prime(N))

